mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java

```diff
@@ -122,17 +132,17 @@ public void setServerSocketFactory(ServerSocketFactory serverSocketFactory) {
   }
 
   public int getPort() {
-    if (port == -1) throw new IllegalStateException("Call play() before getPort()");
+    startOrThrow();
```

Comment:
The explicit life cycle is maintained for non-rule usage. I think we can
kill this though.
On Jan 2, 2015 9:34 PM, "Jesse Wilson" <notifications@github.com> wrote:

> In
> mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java
> <https://github.com/square/okhttp/pull/1277#discussion-diff-22429747>:
>
> > @@ -122,17 +132,17 @@ public void setServerSocketFactory(ServerSocketFactory serverSocketFactory) {
> >    }
> >
> >    public int getPort() {
> > -    if (port == -1) throw new IllegalStateException("Call play() before getPort()");
> > +    startOrThrow();
>
> awww, does it really have to be this automatic? I liked the explicit
> lifecycle
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/okhttp/pull/1277/files#r22429747>.
>